src/epj.ProgressBar.Maui/ProgressBar.cs

```diff
@@ -159,6 +173,72 @@ private void DrawProgress()
             progressPaint.Color = ProgressColor.ToSKColor();
         }
 
+        if (UseSegment && SegmentCount > 0)
+        {
+            using SKPaint basePaint = GenerateSegment(progressPaint);
+        }
+        else
+        {
+            using SKPath progressPath = GenerateProgress(progressPaint);
+        }
+    }
+
+    private SKPath GenerateProgress(SKPaint progressPaint)
+    {
+        var progressPath = new SKPath();
+
+        var progressRect = UseRange
+            ? new SKRect(_info.Width * LowerRangeValue, 0, _info.Width * UpperRangeValue, _info.Height)
+            : new SKRect(0, 0, _info.Width * Progress, _info.Height);
+
+        if (RoundCaps)
+        {
+            progressPath.AddRoundRect(progressRect, progressRect.Height / 2, progressRect.Height / 2);
+        }
+        else
+        {
+            progressPath.AddRect(progressRect);
+        }
+
         _canvas.DrawPath(progressPath, progressPaint);
+        return progressPath;
+    }
+
+    private SKPaint GenerateSegment(SKPaint progressPaint)
```

Comment:
Likewise, the return value of this method is never used except for storing it in a temporary variable. This shouldn't be necessary, you could instead do the following:

```c#
using var basePaint = new SKPaint
{
    Style = SKPaintStyle.Fill,
    IsAntialias = true,
    Color = BaseColor.ToSKColor()
};
```

That way, you don't need to return it, it's cleaned up after being drawn with `_canvas.DrawPath(segmentPath, basePaint);`